composer/loggers/neptune_logger.py

```diff
@@ -0,0 +1,250 @@
+
+    For more, see the [Neptune-Composer integration guide](https://docs.neptune.ai/integrations/composer/).
+    """
+    METRIC_NAMESPACE = 'metrics'
+    HYPERPARAM_NAMESPACE = 'hyperparameters'
+    TRACE_NAMESPACE = 'traces'
+    INTEGRATION_VERSION_KEY = 'source_code/integrations/neptune-MosaicML'
+
+    def __init__(
+        self,
+        *,
+        project: Optional[str] = None,
+        api_token: Optional[str] = None,
+        rank_zero_only: bool = True,
+        upload_artifacts: bool = False,
+        base_namespace: str = 'training',
+        **neptune_kwargs,
+    ) -> None:
+        try:
+            from neptune.internal.utils import verify_type
+        except ImportError as e:
+            raise MissingConditionalImportError(extra_deps_group='neptune',
+                                                conda_package='neptune',
+                                                conda_channel='conda-forge') from e
+
+        verify_type('project', project, (str, type(None)))
+        verify_type('api_token', api_token, (str, type(None)))
+        verify_type('rank_zero_only', rank_zero_only, bool)
+        verify_type('upload_artifacts', upload_artifacts, bool)
+        verify_type('base_namespace', base_namespace, str)
+
+        if not base_namespace:
+            raise ValueError("Argument 'base_namespace' cannot be an empty string.")
+
+        self._project = project
+        self._api_token = api_token
+        self._rank_zero_only = rank_zero_only
+        self._upload_artifacts = upload_artifacts
+        self._base_namespace = base_namespace
+        self._neptune_kwargs = neptune_kwargs
+
+        mode = self._neptune_kwargs.pop('mode', 'async')
+
+        self._enabled = (not rank_zero_only) or dist.get_global_rank() == 0
+
+        self._mode = mode if self._enabled else 'debug'
+
+        self._neptune_run = None
+        self._base_handler = None
+
+        super(NeptuneLogger, self).__init__()
+
+    @property
+    def neptune_run(self):
+        """Gets the Neptune run object from a NeptuneLogger instance.
+
+        You can log additional metadata to the run by accessing a path inside the run and assigning metadata to it
+        with "=" or [Neptune logging methods](https://docs.neptune.ai/logging/methods/).
+
+        Example:
+            >>> neptune_logger = NeptuneLogger()
+            >>> trainer = Trainer(loggers=neptune_logger, ...)
+            >>> trainer.fit()
+            >>> neptune_logger.neptune_run["some_metric"] = 1
+            >>> trainer.close()
+        """
+        from neptune import Run
+
+        if not self._neptune_run:
+            self._neptune_run = Run(
+                project=self._project,
+                api_token=self._api_token,
+                mode=self._mode,
+                **self._neptune_kwargs,
+            )
+        return self._neptune_run
+
+    @property
+    def base_handler(self):
+        return self.neptune_run[self._base_namespace]
```

Comment:
Docstring for `base_handler` property

```suggestion
    def base_handler(self):
        """Gets a handler for the base logging namespace.
    
        Use the handler to log extra metadata to the run and organize it under the base namespace (default: "training").
        You can operate on it like a run object: Access a path inside the handler and assign metadata to it with "=" or
        other [Neptune logging methods](https://docs.neptune.ai/logging/methods/).
    
        Example:
            >>> neptune_logger = NeptuneLogger()
            >>> trainer = Trainer(loggers=neptune_logger, ...)
            >>> trainer.fit()
            >>> neptune_logger.base_handler["some_metric"] = 1
            >>> trainer.close()
            Result: The value `1` is organized under "training/some_metric" inside the run.
        """
        return self.neptune_run[self._base_namespace]
```